### PR TITLE
Add version api

### LINKED ISSUE #2924: Paddle Python module does not provide version info.
Just like other python module for example,
import bumpy as np
print np.__version__
it will return its version as 1.12.1

so suggest that paddle python module add this version info
import paddle.v2 as paddle
print paddle.__version__
it will return some version information
it is a common practice.

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #2924 
We can use the version API as:
```bash 
>>> print paddle.version.version
0.10.0
>>> print paddle.version.git_commit
e1ac448c3320d8a11051fad63cc17f3803862236
```

### CODE REVIEW

**Comment by wangkuiyi** [comment]
I have a serious concern about our current versioning approach, which takes all Git commits between 0.9.0 and 0.10.0 as 0.9.0. This is not realistic.

The correct logic should be

```python
if paddle.version.tagged:
   print paddle.version # 0.10.0, 0.10.0rc1 or something
   print paddle.version.major # 0
   print paddle.version.minor # 10
   print paddle.version.release # 0
   print paddle.version.rc # 1, or 0 if not a release candidate
else:
   print paddle.version     # 0x1232131abdd3  
   print paddle.version.id # 0x1232131abdd3
```

**Comment by Yancey1989** [comment]
Thanks for @wangkuiyi , to distinguish `taged` or nor is a good idea, I only have some small doubt:
1. `paddle.version ` is a Python module, so we can use `paddle.version.full_version`?
1. Following [Semantic Versioning 2.0.0](http://semver.org/#semantic-versioning-200), the version number is made up by MAJOR.MINOR.PATCH
1. If not a `taged` version, maybe we only print the commit? 

So how about the following logic?

```python
if paddle.version.tagged:
   print paddle.version.full_version         # 0.10.0, 0.10.0rc1 or something
   print paddle.version.major   # 0
   print paddle.version.minor   # 10
   print paddle.version.patch # 0
   print paddle.version.rc      # 1, or 0 if not a release candidate
else:
   print paddle.version.commit # 1232131abdd3
```

**Comment by wangkuiyi** [comment]
@Yancey1989 Completely agree! Thanks!

**Comment by Yancey1989** [comment]
Done with @wangkuiyi 's comments.

File: doc/design/releasing_process.md [hunk 1]
```diff
@@ -20,17 +21,17 @@ PaddlePaddle每次发新的版本，遵循以下流程:
...
+1. 协同完成Release Note的书写
 
 
 需要注意的是:
 
 * `release/版本号`分支一旦建立，一般不允许再从`develop`分支合入`release/版本号`。这样保证`release/版本号`分支功能的封闭，方便测试人员测试PaddlePaddle的行为。
 * 在`release/版本号`分支存在的时候，如果有bugfix的行为，需要将bugfix的分支同时merge到`master`, `develop`和`release/版本号`这三个分支。
 
-# PaddlePaddle 分支规范
+## PaddlePaddle 分支规范
```
Comment (Yancey1989) [inline doc/design/releasing_process.md]: With markdownlint [MD025](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md025---multiple-top-level-headers-in-the-same-document), there is only one top-level header.

**Comment by typhoonzero** [comment]
Seem CI failed at:
```
File "setup.py", line 46, in write_version_py
with open(filename, 'w') as f:
IOError: [Errno 2] No such file or directory: '$/paddle/python/paddle/version.py'
```

**Comment by Yancey1989** [comment]
@typhoonzero , I'm debugging with this failed and will update this PR later.

**Pull request merged**